src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -1546,25 +1559,26 @@ bool CompileBroker::compilation_is_in_queue(const methodHandle& method) {
 // CompileBroker::compilation_is_prohibited
 //
 // See if this compilation is not allowed.
-bool CompileBroker::compilation_is_prohibited(const methodHandle& method, int osr_bci, int comp_level, bool excluded) {
+bool CompileBroker::compilation_is_prohibited(const methodHandle& method, int osr_bci, int comp_level) {
+  assert(compiler(comp_level) != nullptr, "Ensure we have a compiler");
```

Comment:
Are you sure `compiler(comp_level)` can never be `nullptr` (not even potentially)? (it seems so to me too but just to make sure)

Reply:
Yes, I am sure.
This is a private member function which is only used in `CompileBroker::compile_method`.
In `CompileBroker::compile_method`  there has always been an assert checking availability of a compiler. I added the assert here just in case the function is used in any other functions where a compiler might not be available.

If we want to minimize changes, we can keep the original code where the  parameter `excluded` is only deleted. The checks `comp == nullptr` are redundant in the original code because the function is invoked as follows:
```
if (comp == nullptr || compilation_is_prohibited(...)) {
    return nullptr;
}
```

I can add a check `comp == nullptr` at the beginning of the function and return `nullptr` if it is true.